### PR TITLE
EP-3985: add maxSpatialResolution parameter for Sentinelhub pyramid c…

### PR DESCRIPTION
**Pull request by JeroenVerstraelen**
Add maxSpatialResolution parameter when creating Sentinelhub pyramid.
This PR is linked with [the one in the openeo-geopyspark-driver project](https://github.com/Open-EO/openeo-geopyspark-driver/pull/104).

### CODE REVIEW

**Comment by jdries** [comment]
There's a compilation error?

**Comment by JeroenVerstraelen** [comment]
My bad, those errors should now be gone.

File: geotrellis-sentinelhub/src/main/scala/org/openeo/geotrellissentinelhub/PyramidFactory.scala [hunk 1]
```diff
@@ -47,9 +46,9 @@ object PyramidFactory {
   // convenience method for Python client
   // TODO: change name? A 429 response makes it rate-limited anyway.
   def rateLimited(endpoint: String, collectionId: String, datasetId: String, clientId: String, clientSecret: String,
-                  processingOptions: util.Map[String, Any], sampleType: SampleType): PyramidFactory =
+                  processingOptions: util.Map[String, Any], sampleType: SampleType, maxSpatialResolution: CellSize = CellSize(10,10)): PyramidFactory =
```
Comment (jdries) [inline geotrellis-sentinelhub/src/main/scala/org/openeo/geotrellissentinelhub/PyramidFactory.scala:49]: Is this going to be backwards compatible when called from python? I'm not sure if the default arguments work in there.

Comment (JeroenVerstraelen) [reply geotrellis-sentinelhub/src/main/scala/org/openeo/geotrellissentinelhub/PyramidFactory.scala]: The default arguments should be backwards compatible now. The build fails due to the same two tests that are broken in the develop branch: 
- org.openeo.geotrellissentinelhub.S3ServiceTest.download_stac_dataCanHandleBatchJobRetries
- org.openeo.geotrellissentinelhub.S3ServiceTest.download_stac_data

We should be able to merge without issues.

Comment (jdries) [reply geotrellis-sentinelhub/src/main/scala/org/openeo/geotrellissentinelhub/PyramidFactory.scala]: Great!

**Pull request merged**